### ISSUE TITLE
Support for always upper case acronym

### ISSUE THREAD

**Issue by davidt99** [JoelLefkowitz/convert-case#3]
There are certain words, usually acronyms, that are always upper-case, for example, IP, FQDN and etc.
When trying to change the case, I'm getting MixCase error.
I would imagine a flag being sent to support this, where a sequence of upper-case characters are considered a word and then according to the flag, we will do:
1. raise (current behavior)
2. upper: keep it as upper
3. lower: treat it as a word, but lower it
4. capital: only keep the first letter as upper

Let me know what you think

**Comment by JoelLefkowitz** [comment]
Hey @davidt99 thanks for raising this. It's a really good question. I don't think we should ignore any sequence of upper case letters since there are common acronyms that are also words that may appear in a sentence. For example I would expect `lower_case("DOING IT ASAP")` to produce: `"doing it asap"` not `"doing it ASAP"`.

One way that might be more suitable would be to add an options parameter to each conversion function that lists the acronyms to ignore:

```
lower_case("DOING IT ASAP", {"acronyms": ["ASAP"]})
> "doing it ASAP"
```

An option I think we should avoid is adding global state like a list of acronyms to ignore. This is because it makes the library harder to extend (and not suitable for concurrency) by obfuscating the implementation details.

What do you think?

**Comment by davidt99** [comment]
I agree we shouldn't have a global list, but passing a list of acronyms to ignore poses the same challenge: coming up with the list :).
Since the convert-case is used in many places, passing the same list will result in a global list, just on the user's side.

**Comment by JoelLefkowitz** [comment]
Looks like there are two issues to solve: Which acronyms to ignore and where to store the list. We can provide a common list of acronyms but I think the default should be none. That way it will cause the least surprise to the user. 

For where store them, the reason to avoid having a global list in the library is that it’s not clear from the function signatures that they read from the list, which means it hides the state from the user. Having a global list on the user side that needs to be passed explicitly to each function doesn't seem like a problem to me.

It might be worthwhile opening it up beyond just acronyms and include lower case letters that never change too, maybe calling them invariants. For example: `lower_case("I LIKE PyPI", {"invariants": ["PyPI", "I"]})` would give `"I like PyPI"`
